prometheus/prometheus.go

```diff
@@ -252,11 +252,16 @@ var forbiddenChars = regexp.MustCompile("[ .=\\-/]")
 
 func flattenKey(parts []string, labels []metrics.Label) (string, string) {
 	key := strings.Join(parts, "_")
-	key = forbiddenChars.ReplaceAllString(key, "_")
+
+	// Only run the regex replace if the key contains forbidden characters to
+	// void overhead on well-formed keys.
+	if strings.ContainsAny(key, " .=-/") {
+		key = forbiddenChars.ReplaceAllString(key, "_")
```

Comment:
I don't know if it's worth the time to try it, but it should only be a few minutes...

I wonder if it would be better to just remove the regex entirely and use [`Replacer`](https://pkg.go.dev/strings#Replacer) instead. My guess is that would be just as efficient as the `contains` call you added above even in the case where nothing is matched and more efficient even when there are replacements to be made as well as "the right tool for the job" from the stdlib?

Reply:
Hey @banks and thanks for the suggestion and the snippet of knowledge around the replacer function. I have updated the change to replace the regex with a replacer which performs on-par/better than skipping the regex with well-formed keys, and much better with keys that contain forbidden characters.